app/lib/test_applications.rb

```diff
@@ -11,6 +12,8 @@ def self.generate_for_provider(provider:, courses_per_application:, count:)
   end
 
   def self.create_application(states:, courses_to_apply_to: nil)
+    raise ZeroCoursesPerApplicationError.new('You can\'t have zero courses per application') unless states.any?
```

Comment:
You could use `blank?` here if you wanted to handle both `nil` and `[]` values with the same error. I don't think this is an issue because we are not passing `nil` to this method from anywhere as far as I can see.

Reply:
Hey Steve thanks for the pointer - I think you're right in this case it shouldn't matter, but it is good to know that this could have be an option in future in similar circumstances 👍